include/openssl/core_dispatch.h

```diff
@@ -19,6 +19,13 @@
 extern "C" {
 # endif
 
+/*
+ * Generic function pointer for provider method arrays, or other contexts where
+ * functions of various signatures must occupy a common slot in an array of
+ * structures.
+ */
+typedef void (*OSSL_FUNC)(void);
```

Comment:
Why did you add this just for the load dispatch table entry?

Reply:
It is used broadly in the ML-KEM case, and already slated to go in.  This just does the same thing in the ML-DSA branch, which will just use the same construct when both are merged.